src/riak_core_app.erl

```diff
@@ -152,8 +152,11 @@ register_capabilities() ->
 %% TODO: This belongs in riak_kv - an issue will be created to move it, but time constraints.
 start_eleveldb_info_service() ->
     Registration = {eleveldb, set_metadata_pid, []},
-    Shutdown = {?MODULE, start_eleveldb_info_service, []},
+    Shutdown = {?MODULE, shutdown_eleveldb_info_service, []},
```

Comment:
Changing this to `{eleveldb, remove_metadata_pid, []}` should shut xref and dialyzer up, with the added benefit of being consistent with registration and eliminating the separate `shutdown_eleveldb_info_service/1` function altogether.